node/test/polkadot-simnet/common/src/lib.rs

```diff
@@ -0,0 +1,353 @@
+			.into_iter()
+			.filter_map(|event| {
+				match event.event {
+					Event::TechnicalCommittee(TechnicalCollectiveEvent::Proposed(_, index, hash, _)) => Some((index, hash)),
+					_ => None
+				}
+			})
+			.next()
+			.ok_or_else(|| "failed to execute council::Call::propose(democracy::Call::fast_track))")?;
+
+		// vote
+		for member in &technical_collective[1..] {
+			let call = TechnicalCollectiveCall::vote(hash.clone(), index, true);
+			node.submit_extrinsic(call, member.clone()).await?;
+		}
+		node.seal_blocks(1).await;
+
+		// close vote
+		let call = TechnicalCollectiveCall::close(hash, index, weight, length);
+		node.submit_extrinsic(call, technical_collective[0].clone()).await?;
+		node.seal_blocks(1).await;
+
+		// assert that fast-track proposal has been passed on chain
+		let events = node.events()
+			.into_iter()
+			.filter(|event| {
+				match event.event {
+					Event::TechnicalCommittee(TechnicalCollectiveEvent::Closed(_, _, _)) |
+					Event::TechnicalCommittee(TechnicalCollectiveEvent::Approved(_)) |
+					Event::TechnicalCommittee(TechnicalCollectiveEvent::Executed(_, Ok(()))) => true,
+					_ => false,
+				}
+			})
+			.collect::<Vec<_>>();
+
+		// make sure all 3 events are in state
+		assert_eq!(events.len(), 3);
+	}
+
+	// now runtime upgrade proposal is a fast-tracked referendum we can vote for.
+	let referendum_index = node.events()
+		.into_iter()
+		.filter_map(|event| match event.event {
+			Event::Democracy(democracy::Event::<Runtime>::Started(index, _)) => Some(index),
+			_ => None,
+		})
+		.next()
+		.ok_or_else(|| "failed to execute council::Call::close")?;
+	let call = DemocracyCall::vote(
+		referendum_index,
+		AccountVote::Standard {
+			vote: Vote { aye: true, conviction: Conviction::Locked1x },
+			// 10 DOTS
+			balance: 10_000_000_000_000,
+		},
+	);
+	for whale in whales {
+		node.submit_extrinsic(call.clone(), whale).await?;
+	}
+
+	// wait for fast track period.
+	node.seal_blocks(FastTrackVotingPeriod::get() as usize).await;
+
+	// assert that the proposal is passed by looking at events
+	let events = node.events()
+		.into_iter()
+		.filter(|event| {
+			match event.event {
+				Event::Democracy(democracy::Event::Passed(_)) |
+				Event::Democracy(democracy::Event::PreimageUsed(_, _, _)) |
+				Event::Democracy(democracy::Event::Executed(_, true)) => true,
+				_ => false,
+			}
+		})
+		.collect::<Vec<_>>();
+
+	// make sure all events were emitted
+	assert_eq!(events.len(), 3);
+	Ok(())
+}
+
+/// Runs the test-runner as a binary.
+pub fn run<F, Fut>(callback: F) -> Result<(), Box<dyn Error>>
+	where
+		F: FnOnce(Node<PolkadotChainInfo>) -> Fut,
+		Fut: Future<Output=Result<(), Box<dyn Error>>>,
+{
+	use structopt::StructOpt;
+	use sc_cli::{CliConfiguration, SubstrateCli};
+
+	let mut tokio_runtime = build_runtime()?;
+	let task_executor = task_executor(tokio_runtime.handle().clone());
+	// parse cli args
+	let cmd = <polkadot_cli::Cli as StructOpt>::from_args();
```

Comment:
There should be variant of this `run` that allows passing the `Cli` args explicitly rather than expecting them to present in the environment.

Reply:
hmm, interesting but what would be the use case?

Reply:
My idea was to also use this for small scale integration test level networks, launching 3 nodes i.e. which should be smth a regular dev laptop can handle.

Reply:
could be added in a future PR, when the need for the functionality arises